include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -834,6 +834,173 @@ class __static_monotonic_dispatcher<::std::integer_sequence<::std::uint16_t, _X,
     }
 };
 
+// For use with __lazy_ctor_storage
+struct __lazy_load_transform_op
+{
+    template <typename _IdxType1, typename _IdxType2, typename _SourceAcc, typename _DestAcc>
+    void
+    operator()(_IdxType1 __idx_source, _IdxType2 __idx_dest, _SourceAcc __source_acc, _DestAcc __dest_acc) const
+    {
+        __dest_acc[__idx_dest].__setup(__source_acc[__idx_source]);
+    }
+};
+
+template <std::uint8_t __vec_size>
+struct __vector_load
+{
+    static_assert(__vec_size <= 4, "Only vector sizes of 4 or less are supported");
+    std::size_t __n;
+    template <typename _IdxType, typename _LoadOp, typename... _Acc>
+    void
+    operator()(std::true_type, _IdxType __start_idx, _LoadOp __load_op, _Acc... __acc) const
+    {
+        _ONEDPL_PRAGMA_UNROLL
+        for (std::uint8_t __i = 0; __i < __vec_size; ++__i)
+            __load_op(__start_idx + __i, __i, __acc...);
+    }
+
+    template <typename _IdxType, typename _LoadOp, typename... _Acc>
+    void
+    operator()(std::false_type, _IdxType __start_idx, _LoadOp __load_op, _Acc... __acc) const
+    {
+        std::uint8_t __elements = std::min(std::size_t{__vec_size}, std::size_t{__n - __start_idx});
```

Comment:
We assume here that `std::min(std::size_t{}, std::size_t{})` will always fit into `std::uint8_t` type?